jira_agile_metrics/calculators/cycletime.py

```diff
@@ -237,25 +235,28 @@ def calculate_cycle_times(
                 impediment_flag = None
                 impediment_start = None
                 impediment_start_status = None
-            
-            # Wipe timestamps if items have moved backwards; calculate cycle time
+
+            # calculate cycle time
 
             previous_timestamp = None
-            accepted_timestamp = None
-            completed_timestamp = None
+            committed_timestamp = None
+            done_timestamp = None
 
-            for cycle_name in cycle_names:
+            for cycle_name in reversed(cycle_names):
                 if item[cycle_name] is not None:
                     previous_timestamp = item[cycle_name]
 
-                    if accepted_timestamp is None and previous_timestamp is not None and cycle_name in accepted_steps:
-                        accepted_timestamp = previous_timestamp
-                    if completed_timestamp is None and previous_timestamp is not None and cycle_name in completed_steps:
-                        completed_timestamp = previous_timestamp
+                if previous_timestamp is not None:
+                    item[cycle_name] = previous_timestamp
+                    if cycle_name == done_column:
```

Comment:
If I'm thinking about this correctly, one implication of this change (basically, using an `==` instead of an `in` check) and the change to how we identify the "start" and "end" column(s) is that before there could in theory be more than one "accepted" (my language) / "committed' (your language - which is better, I agree) column, and now there can be only one. And the same for "completed"/"done". Correct?

I think that's probably fine. It relates to the point above about trying to design for there potentially being multiple columns with `type==StatusTypes.accepted` (and `complete`), but I'm not really sure that's a valid use case. I think it's an overhang from before I implemented mapping of multiple JIRA statuses to a single logical workflow step in this tool.

It does - and I think you said this before - mean that if you have multiple things after "done" *and* the ticket never touches the (single) "done" state but skips straight to one of the later ones, then it'll appear to not have been completed. Again, that's a slight change of behaviour but I'm not sure it massively matters as it seems unlikely people will have this happen.

Do you agree?

Reply:
See the picture in https://github.com/DeloitteDigitalUK/jira-agile-metrics/pull/27#issuecomment-636308642, all workflow stages starting from `Committed column` until and excluding `Done column` are `active_columns`. Workflow stages after `Done column` can be ignored.